EventListener/TransactionalListener.php

```diff
@@ -177,7 +189,23 @@ public function onKernelView(GetResponseForControllerResultEvent $event)
                     EmbeddedManager::GROUP_ALL
                 );
             }
+        }
+    }
 
+    /**
+     * On kernel response
+     *
+     * @param FilterResponseEvent $event
+     *
+     * @return void
+     */
+    public function onKernelResponse(FilterResponseEvent $event)
+    {
+        $request = $event->getRequest();
+        $transaction = $request->attributes->get('transaction');
+        if ($transaction) {
+            $this->transactionUpdater->updateResponseTime($transaction, $request, microtime(true));
```

Comment:
You can move this update to the end of the `if ($transaction instanceof Transaction)` block in onKernelView above and avoid modifying the json response directly. It may not be as accurate of a representation of the full response time but we will have the full response content so I think it's sufficient.